languages/thingtalk/dialogue_acts/action-confirm.js

```diff
@@ -68,7 +72,37 @@ function actionConfirmAcceptPhrase(ctx) {
     return clone.state;
 }
 
+function actionConfirmRejectPhrase(ctx) {
+    return new Ast.DialogueState(null, POLICY_NAME, 'cancel', null, []);
+}
+
+function actionConfirmChangeParam(ctx, answer) {
+    const questions = ctx.dialogueActParam || [];
+    if (answer instanceof Ast.Value) {
+        if (questions.length !== 1)
+            return null;
+        answer = new Ast.InputParam(null, questions[0], answer);
+    }
+
+    const action = C.getInvocation(ctx.state.history[0]);
```

Comment:
You don't want to look at the first result in the history, you want to look at the next one (`ctx.next`).